dateparser/date.py

```diff
@@ -282,33 +282,32 @@ class DateDataParser(object):
     :raises:
             ValueError - Unknown Language, TypeError - Languages argument must be a list
     """
-    language_loader = None
+    available_language_map = {}
 
     @apply_settings
     def __init__(self, languages=None, allow_redetect_language=False, settings=None):
         self._settings = settings
-        available_language_map = self._get_language_loader().get_language_map()
-
-        if isinstance(languages, (list, tuple, collections.Set)):
-
-            if all([language in available_language_map for language in languages]):
-                languages = [available_language_map[language] for language in languages]
+        self.allow_redetect_language=allow_redetect_language
+        self.languages=languages
+        self.are_languages_passed = bool(languages)
+
+        available_language_shortnames=['en', 'ar', 'be', 'bg', 'bn', 'cs', 'da', 'de', 'es', 
```

Comment:
We must not mix data loading logic with this class here especially when we can completely avoid it by making necessary changes in `LanguageDataLoader` class.